### PR TITLE
 [aci_epg_to_contract_interface] Added new resource fv_rs_cons_if .

### PR DESCRIPTION
**Pull request by anvitha-jain**
Made changes to examples files so that they can run on their own. solves #812

### CODE REVIEW

File: website/docs/r/fvrsconsif.html.markdown [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+# aci_contract_interface #
+
+Manages ACI Contract Interface
+
+## API Information ##
+
+* `Class` - fvRsConsIf
+* `Distinguished Name` - uni/tn-{name}/ap-{name}/epg-{name}/rsconsIf-{tnVzCPIfName}
```
Comment (lhercot) [inline website/docs/r/fvrsconsif.html.markdown:16]: tnVzCPIfName -> contract_interface_name

File: website/docs/d/fvrsconsif.html.markdown [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# aci_contract_interface #
+
+Data source for ACI Contract Interface
+
+
+## API Information ##
+
+* `Class` - fvRsConsIf
+* `Distinguished Name` - uni/tn-{name}/ap-{name}/epg-{name}/rsconsIf-{tnVzCPIfName}
```
Comment (lhercot) [inline website/docs/d/fvrsconsif.html.markdown:17]: tnVzCPIfName -> contract_interface_name

File: website/docs/r/fvrsconsif.html.markdown [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+* `Location` - Tenant -> Application Profiles -> Application EPGs -> Contracts
+
+
+## Example Usage ##
+
+```hcl
+resource "aci_contract_interface" "example" {
+  application_epg_dn  = aci_application_epg.example.id
+  tnVzCPIfName  = "example"
```
Comment (lhercot) [inline website/docs/r/fvrsconsif.html.markdown:28]: Update this too

File: website/docs/r/fvrsconsif.html.markdown [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  prio = "unspecified"
+
+}
+```
+
+## Argument Reference ##
+
+* `application_epg_dn` - (Required) Distinguished name of the parent ApplicationEPG object.
+* `tnVzCPIfName` - (Required) TnVzCPIfName of the object Contract Interface.
```
Comment (lhercot) [inline website/docs/r/fvrsconsif.html.markdown:37]: Update this too

**Review by lhercot — CHANGES_REQUESTED** [review]

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  application_epg_dn  = aci_application_epg.example.id
+  contract_interface_dn = aci_imported_contract.contract_interface.id
+  prio = "unspecified"
+
+}
+```
+
+## Argument Reference ##
+
+* `application_epg_dn` - (Required) Distinguished name of the parent ApplicationEPG object.
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:37]: Application EPG

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+* `contract_interface_dn` - (Required) Distinguished name of the object Contract Interface.
+* `annotation` - (Optional) Annotation of the object Contract Interface.
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:39]: of the Contract Interface Relationship object

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  contract_interface_dn = aci_imported_contract.contract_interface.id
+  prio = "unspecified"
+
+}
+```
+
+## Argument Reference ##
+
+* `application_epg_dn` - (Required) Distinguished name of the parent ApplicationEPG object.
+* `contract_interface_dn` - (Required) Distinguished name of the object Contract Interface.
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:38]: of the Contract Interface object

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+* `annotation` - (Optional) Annotation of the object Contract Interface.
+* `prio` - (Optional) prio.The contract interface priority. Allowed values are "level1", "level2", "level3", "level4", "level5", "level6", "unspecified", and default value is "unspecified". Type: String.
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:40]: (Optional) The Contract Interface Relationship priority. ...

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+
+## Importing ##
+
+An existing ContractInterface can be [imported][docs-import] into this resource via its Dn, via the following command:
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:45]: An existing Contract Interface Relationship object can be...

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+[docs-import]: https://www.terraform.io/docs/import/index.html
+
+
+```
+terraform import aci_contract_interface.example <Dn>
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:50]: aci_epg_to_contract_interface

File: website/docs/d/epg_to_contract_interface.html.markdown [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+layout: "aci"
+page_title: "ACI: aci_contract_interface"
+sidebar_current: "docs-aci-data-source-contract_interface"
+description: |-
+  Data source for ACI Contract Interface
+---
+
+# aci_contract_interface #
+
+Data source for ACI Contract Interface
```
Comment (lhercot) [inline website/docs/d/epg_to_contract_interface.html.markdown:12]: for ACI Contract Interface Relationship

File: website/docs/d/epg_to_contract_interface.html.markdown [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+## GUI Information ##
+
+* `Location` - Tenant -> Application Profiles -> Application EPGs -> Contracts
+
+
+
+## Example Usage ##
+
+```hcl
+data "aci_contract_interface" "example" {
```
Comment (lhercot) [inline website/docs/d/epg_to_contract_interface.html.markdown:29]: aci_epg_to_contract_interface

File: website/docs/d/epg_to_contract_interface.html.markdown [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+  application_epg_dn  = aci_application_epg.example.id
+  contract_interface_dn = aci_imported_contract.contract_interface.id
+}
+```
+
+## Argument Reference ##
+
+* `application_epg_dn` - (Required) Distinguished name of parent ApplicationEPG object.
```
Comment (lhercot) [inline website/docs/d/epg_to_contract_interface.html.markdown:37]: Application EPG

File: website/docs/d/epg_to_contract_interface.html.markdown [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+* `contract_interface_dn` - (Required) Distinguished name of object Contract Interface.
```
Comment (lhercot) [inline website/docs/d/epg_to_contract_interface.html.markdown:38]: of the Contract Interface object

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_fvrsconsif.go [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+	dn := d.Id()
+	d.SetId(fvRsConsIf.DistinguishedName)
+	if dn != fvRsConsIf.DistinguishedName {
+		d.Set("application_epg_dn", "")
+	}
+	fvRsConsIfMap, err := fvRsConsIf.ToMap()
+	if err != nil {
+		return d, err
+	}
+	d.Set("application_epg_dn", GetParentDn(dn, fmt.Sprintf("/"+models.RnfvRsConsIf, fvRsConsIfMap["tnVzCPIfName"])))
```
Comment (lhercot) [inline aci/resource_aci_fvrsconsif.go:78]: add spaces around the +

File: aci/resource_aci_fvrsconsif.go [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+	}
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciContractInterfaceRelationshipCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] ContractInterfaceRelationship: Beginning Creation")
+	aciClient := m.(*client.Client)
+	tnVzCPIfName := GetMOName(d.Get("contract_interface_dn").(string))
+	ApplicationEPGDn := d.Get("application_epg_dn").(string)
```
Comment (lhercot) [inline aci/resource_aci_fvrsconsif.go:105]: applicationEpgDn

File: aci/resource_aci_fvrsconsif.go [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+	d.SetId(fvRsConsIf.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+	return resourceAciContractInterfaceRelationshipRead(ctx, d, m)
+}
+
+func resourceAciContractInterfaceRelationshipUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] ContractInterfaceRelationship: Beginning Update")
+	aciClient := m.(*client.Client)
+	tnVzCPIfName := GetMOName(d.Get("contract_interface_dn").(string))
+	ApplicationEPGDn := d.Get("application_epg_dn").(string)
```
Comment (lhercot) [inline aci/resource_aci_fvrsconsif.go:138]: applicationEpgDn

File: aci/resource_aci_rtctrlattrp.go [hunk 18]
```diff
@@ -124,7 +124,7 @@ func resourceAciActionRuleProfileCreate(ctx context.Context, d *schema.ResourceD
 	if NameAlias, ok := d.GetOk("name_alias"); ok {
 		rtctrlAttrPAttr.NameAlias = NameAlias.(string)
 	}
-	rtctrlAttrP := models.NewActionRuleProfile(fmt.Sprintf("attr-%s", name), TenantDn, desc, rtctrlAttrPAttr)
+	rtctrlAttrP := models.NewActionRuleProfile(fmt.Sprintf("attr-%s", name), TenantDn, desc, "", rtctrlAttrPAttr)
```
Comment (lhercot) [inline aci/resource_aci_rtctrlattrp.go:128]: This should be fixed on latest master. Remove that change and rebase on top of master.

File: website/docs/d/epg_to_contract_interface.html.markdown [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+```hcl
+data "aci_epg_to_contract_interface" "example" {
+  application_epg_dn  = aci_application_epg.example.id
+  contract_interface_dn = aci_imported_contract.contract_interface.id
+}
+```
+
+## Argument Reference ##
+
+* `application_epg_dn` - (Required) Distinguished name of parent Application EPG object.
```
Comment (lhercot) [inline website/docs/d/epg_to_contract_interface.html.markdown:37]: of the parent

File: website/docs/d/epg_to_contract_interface.html.markdown [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+* `contract_interface_dn` - (Required) Distinguished name of object Contract Interface object.
```
Comment (lhercot) [inline website/docs/d/epg_to_contract_interface.html.markdown:38]: of the Contract...

File: website/docs/d/epg_to_contract_interface.html.markdown [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Contract Interface Relationship object.
+* `annotation` - (Optional) Annotation of object Contract Interface Relationship object.
```
Comment (lhercot) [inline website/docs/d/epg_to_contract_interface.html.markdown:42]: of the Contract...

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+  contract_interface_dn = aci_imported_contract.contract_interface.id
+  prio = "unspecified"
+
+}
+```
+
+## Argument Reference ##
+
+* `application_epg_dn` - (Required) Distinguished name of the parent Application EPG object.
+* `contract_interface_dn` - (Required) Distinguished name of the object Contract Interface object.
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:38]: remove first object
the Contract...

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+* `annotation` - (Optional) Annotation of the object Contract Interface Relationship object.
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:39]: remove first object
the Contract...

File: website/docs/r/epg_to_contract_interface.html.markdown [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+* `prio` - (Optional) The contract interface priority. Allowed values are "level1", "level2", "level3", "level4", "level5", "level6", "unspecified", and default value is "unspecified". Type: String.
```
Comment (lhercot) [inline website/docs/r/epg_to_contract_interface.html.markdown:40]: (Optional) The Contract Interface Relationship priority. Allowed...

**Review by lhercot — CHANGES_REQUESTED** [review]

File: website/docs/d/epg_to_contract_interface.html.markdown (hunk 20 above)
Comment (lhercot) [reply website/docs/d/epg_to_contract_interface.html.markdown:38]: this is not fixed

File: website/docs/r/epg_to_contract_interface.html.markdown (hunk 22 above)
Comment (lhercot) [reply website/docs/r/epg_to_contract_interface.html.markdown:38]: still not fixed

**Review by lhercot — CHANGES_REQUESTED** [review]

File: aci/resource_aci_fvrsconsif.go (hunk 15 above)
Comment (anvitha-jain) [reply aci/resource_aci_fvrsconsif.go:78]: fmt doesn't allow this

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**